rapids_pytest_benchmark/rapids_pytest_benchmark/rmm_resource_analyzer.py

```diff
@@ -0,0 +1,59 @@
+import os
+import csv
+import rmm
+import tempfile
+
+
+class RMMResourceAnalyzer:
+    """
+    Class to control enabling, disabling, & parsing RMM resource
+    logs.
+    """
+
+    def __init__(self):
+        self.max_gpu_util = -1
+        self.max_gpu_mem_usage = 0
+        self.leaked_memory = 0
+        log_file_name = "rapids_pytest_benchmarks_log"
+        self._log_file_prefix = os.path.join(tempfile.gettempdir(), log_file_name)
+
+    def enable_logging(self):
+        """
+        Enable RMM logging. RMM creates a CSV output file derived from
+        provided file name that looks like: log_file_prefix + ".devX", where
+        X is the GPU number.
+        """
+        rmm.enable_logging(log_file_name=self._log_file_prefix)
+
+    def disable_logging(self):
+        """
+        Disable RMM logging
+        """
+        log_output_files = rmm.get_log_filenames()
+        rmm.mr._flush_logs()
+        rmm.disable_logging()
+        self._parse_results(log_output_files)
```

Comment:
Maybe a `FIXME` here for this, but we may want to only parse the files that correspond to a specified GPU (if they specified a GPU). Doing this is okay for now though.